game-core/src/main/java/games/strategy/engine/data/GameMap.java

```diff
@@ -207,6 +209,46 @@ public Territory getTerritory(final String s) {
     return getNeighbors(frontier, searched, distance, null);
   }
 
+  /**
+   * Returns all neighbors within a certain distance of the starting territory that territories
+   * between the 2 match the condition. Does NOT include the original/starting territory in the
+   * returned Set.
+   */
+  public Set<Territory> getNeighborsIgnoreEnd(final Territory territory, final int distance,
+      final Predicate<Territory> cond) {
+    if (distance < 0) {
+      throw new IllegalArgumentException("Distance must be positive not: " + distance);
+    }
+    if (distance == 0) {
+      return Collections.emptySet();
+    }
+    final Set<Territory> neighbors = new HashSet<>(getNeighbors(territory));
+    if (distance == 1) {
+      return neighbors;
+    }
+    final Set<Territory> start = getNeighbors(territory, cond);
+    for (int i = 2; i <= distance; i++) {
+      neighbors.addAll(getNeighborsIgnoreEnd(start, new HashSet<>(start), i - 1, cond));
+    }
+    neighbors.remove(territory);
+    return neighbors;
+  }
+
+  private Set<Territory> getNeighborsIgnoreEnd(final Set<Territory> frontier, final Set<Territory> searched,
+      final int distance, @Nullable final Predicate<Territory> cond) {
+    if (distance == 0) {
```

Comment:
Very minor: A comment that this condition terminates recursion could be a good heads-up for the future maintenance reviewer. I was wondering a bit when this could be zero to eventually see it could be after recursive iteration.

Reply:
Updated.